axon-server-connector/src/main/java/org/axonframework/axonserver/connector/event/AxonServerEventStorageEngine.java

```diff
@@ -0,0 +1,181 @@
+     *                   {@link org.axonframework.messaging.MetaData} values into bytes.
+     */
+    public AxonServerEventStorageEngine(@Nonnull AxonServerConnection connection,
+                                        @Nonnull Converter converter) {
+        this.connection = Objects.requireNonNull(connection, "The Axon Server connection cannot be null.");
+        this.converter = new EventConverter(converter);
+    }
+
+    @Override
+    public CompletableFuture<AppendTransaction> appendEvents(@Nonnull AppendCondition condition,
+                                                             @Nonnull List<TaggedEventMessage<?>> events) {
+        DcbEventChannel.AppendEventsTransaction appendEventsTransaction = eventChannel().startTransaction();
+        appendEventsTransaction.condition(ConditionConverter.convertAppendCondition(condition));
+        if (events.isEmpty()) {
+            return CompletableFuture.completedFuture(null);
+        } else {
+            events.stream()
+                  .map(converter::convertTaggedEventMessage)
+                  .forEach(taggedEvent -> {
+                      if (logger.isDebugEnabled()) {
+                          logger.debug("Appended event [{}] with timestamp [{}].",
+                                       taggedEvent.getEvent().getIdentifier(),
+                                       taggedEvent.getEvent().getTimestamp());
+                      }
+                      appendEventsTransaction.append(taggedEvent);
+                  });
+        }
+
+        return CompletableFuture.completedFuture(new AppendTransaction() {
+            @Override
+            public CompletableFuture<ConsistencyMarker> commit() {
+                return appendEventsTransaction.commit()
+                                              .exceptionallyCompose(throwable -> CompletableFuture.failedFuture(
+                                                      new AppendEventsTransactionRejectedException(throwable.getMessage())
+                                              ))
+                                              .thenApply(appendResponse -> new GlobalIndexConsistencyMarker(
+                                                      // TODO we need to add the size of the event's?!
+                                                      appendResponse.getLastPosition() + events.size()
+                                              ));
+            }
+
+            @Override
+            public void rollback() {
+                appendEventsTransaction.rollback();
+            }
+        });
+    }
+
+    @Override
+    public MessageStream<EventMessage<?>> source(@Nonnull SourcingCondition condition) {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Start sourcing events with condition [{}].", condition);
+        }
+
+        // TODO Do we disregard SourcingCondition#end for now?
```

Comment:
We could throw an exception if end is lower than `Long.MAX_VALUE`, and then implement it when Axon Server supports it.